### ISSUE TITLE
Authentication failed

### ISSUE THREAD

**Issue by drunkensqusih** [karimra/gnmic#496]
I have noticed an issue with the  parsing of passwords when they are provided via ENV or Config

Given the following password
```GNMIC_PASSWORD='|Bo|Z%TYe*&$P33~'```

I see this in the debug logs when i start up a subscriber 

```
[config] 2021/11/18 17:25:24.532736 /home/runner/work/gnmic/gnmic/config/targets.go:105: read target config: {"name":"testing","address":"127.0.0.1:50051","username":"automation","password":"|Bo|Z%TYe*\u0026$P33~","timeout":2000000000,"insecure":true,"skip-verify":true,"subscriptions":["interface_stats"],"outputs":["STDOUT"],"retry-timer":10000000000,"gzip":false,"token":""}

[config] 2021/11/18 17:25:24.532759 /home/runner/work/gnmic/gnmic/config/targets.go:119: targets: map[testing:{"name":"testing","address":"127.0.0.1::50051","username":"automation","password":"|Bo|Z%TYe*\u0026~","timeout":2000000000,"insecure":true,"skip-verify":true,"subscriptions":["interface_stats"],"outputs":["STDOUT"],"retry-timer":10000000000,"gzip":false,"token":""}]
```

Not only is it doing some UTF encoding but it actually seems to be removing characters after the `$`

If i try this via CLI it works as expected and this is in the debug output

```
[gnmic] password='|Bo|Z%TYe*&$P33~'(string)
```

**Comment by karimra** [comment]
Hi,

Thanks for catching and reporting this.

The UTF encoding comes from JSON Marshaling the target configuration and is only related to log printing, the password is still stored as it is in ENV or config file.
The real issue is the "disappearing" characters after $, that comes from trying to expand ENV variables in the config.

I did a few tests and it seems that a reasonable fix is: do not expand password fields if its value does not start with `$`

I will push a PR with the fix today.

**Comment by karimra** [comment]
Hi, the fix is part of release v0.21.0.
If you can please give it a try.
